integration-tests/tests/run-notebook.js

```diff
@@ -14,37 +14,19 @@ const testRunNotebookFn = _.flow(
   await click(page, clickable({ textContains: 'View Workspaces' }))
   await signIntoTerra(page, token)
   await dismissNotifications(page)
-  await findElement(page, clickable({ textContains: workspaceName }))
   await waitForNoSpinners(page)
-  await click(page, clickable({ textContains: workspaceName }))
-  await delay(5000)
+
+  await waitForNoSpinnersAfterAction(page, () => click(page, clickable({ textContains: workspaceName })))
   await click(page, navChild('notebooks'))
-  await waitForNoSpinners(page)
   await click(page, clickable({ textContains: 'Create a' }))
   await fillIn(page, input({ placeholder: 'Enter a name' }), notebookName)
   await select(page, 'Language', 'Python 3')
-  await click(page, clickable({ text: 'Create Notebook' }))
-  await click(page, clickable({ textContains: notebookName }))
-  // We are adding a short micro sleep to make sure that the spinner is
-  // rendered by the time we start waiting for the spinner to go away.
-  await delay(500)
-  await waitForNoSpinners(page)
-  await click(page, clickable({ text: 'Edit' }))
-  // There are two separate activities in the UI that could interfere with beginning to interact
-  // with the modal for creating a cloud environment:
-  //   1. AJAX calls to load cloud environment details and available docker images
-  //      - renders a spinner overlay on top of UI elements
-  //   2. The drawer slide-in animation
-  //      - causes UI elements to move before puppeteer clicks on them
-  // Experimentation has shown that there's enough of a gap between clicking 'Edit' and the AJAX
-  // spinner being rendered that simply waiting for no spinners does not work; the test recognizes
-  // that there aren't any spinners before the spinner has a chance to render. Additionally, even
-  // though the slide-in animation is supposedly only 200ms, experimentation has shown that even a
-  // 500ms delay in the test is not enough to guarantee that the UI elements have finished moving.
-  // Therefore, we start with a 2000ms delay, then make sure there are no spinners just in case the
-  // AJAX calls are unexpectedly slow.
-  await delay(2000)
-  waitForNoSpinners(page)
+
+  await waitForNoSpinnersAfterAction(page, () => click(page, clickable({ text: 'Create Notebook' })))
```

Comment:
Each of these actions is followed up by a busy spinner, so I pair taking the action with waiting on the spinner.